src/Results/DTO/GenerativeAiResult.php

```diff
@@ -328,8 +332,9 @@ public function toTexts(): array
         foreach ($this->candidates as $candidate) {
             $message = $candidate->getMessage();
             foreach ($message->getParts() as $part) {
+                $channel = $part->getChannel();
                 $text = $part->getText();
-                if ($text !== null) {
+                if (MessagePartChannelEnum::content() === $channel && $text !== null) {
```

Comment:
Same nit-pick as above, just a suggestion:
```suggestion
                if ($channel->isContent() && $text !== null) {
```